DistroLauncher/named_mutex.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "stdafx.h"
+
+DWORD Win32MutexApi::create(HANDLE& mutex_handle, LPCWSTR mutex_name) noexcept
+{
+    mutex_handle = CreateMutex(nullptr, FALSE, mutex_name);
+    if (mutex_handle == nullptr) {
+        const auto error = GetLastError();
+        return error != 0 ? error
+                          : ~static_cast<DWORD>(0); // Ensuring return value is always != 0 when mutex_handle is nullptr
+    }
+    return 0;
+}
+
+DWORD Win32MutexApi::destroy(HANDLE& mutex_handle, LPCWSTR mutex_name) noexcept
+{
+    if (static_cast<bool>(mutex_handle)) {
```

Comment:
Yeah, that was my idea. But then the compiler raises a warning next line because "mutex_handle might be zero". Evidently not true, but the compiler doesn't notice unless I do this.

Reply:
I can always use `__asume(mutex_handle != 0)`, but then clang-tidy may force me to explicitly cast it, getting back to the same problem 😓

Reply:
`if(mutex_handle != NULL)` then?

Reply:
I'll give it a try.

Reply:
Done